### PR TITLE
Add: Save lawsuit form fields using SharedPreferences. Change: Have messages functions + lists (Db + Inbox) in one place. s/DbMessages/Messages

### PR DESCRIPTION
**Pull request by cotom972**
**SharedPreferences:**
`Lawsuit.proto` holds all of the relevant data needed for a single lawsuit.

When a user begins to create a lawsuit for a specific message, values of the form are passed to a `Lawsuit.proto` which is been saved to the sharedPreferences under a key of: 

`String.valueOf(sms.getSender() + sms.getBody() + sms.getReceivedAt()).hashCode())`

**Change s/DbMessages/Messages:**
Handle messages lists (Inbox and spams from the db) in one dedicated place, as well as functions related to the messages.

Inbox didn't have a dedicated place before.

### CODE REVIEW

File: projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java [hunk 1]
```diff
@@ -1,35 +1,44 @@
 package hasadna.noloan;
 
 import android.content.Intent;
+import android.graphics.Color;
```
Comment (cotom972) [inline projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java]: Not relevant - To be removed after review. As well as other not used imports.

File: projects/noloan/app/src/main/java/hasadna/noloan/lawsuit/LawsuitActivity.java [hunk 2]
```diff
@@ -347,5 +366,39 @@ public void onBackPressed() {
...
+            .edit();
+    editor.putString(
+        sharedPreferencesKey,
+        Base64.encodeToString(this.lawsuitProto.toByteArray(), Base64.DEFAULT));
+    editor.apply();
+  }
+
+  public Lawsuit getSharedPreferencesLawsuitProto() {
+
+    // Check if previous this lawsuit exists in the sharedPreferences
```
Comment (cotom972) [inline projects/noloan/app/src/main/java/hasadna/noloan/lawsuit/LawsuitActivity.java]: ->  `` // Check if this previous lawsuit exists in the sharedPreferences``

File: projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java [hunk 3]
```diff
@@ -152,14 +134,23 @@ public void bind(SmsMessage sms) {
...
     }
   }
 }
 
+/**
+ * // Search message by Body and Sender. Return -1 of no message found public int
+ * searchMessage(SmsMessage smsMessage) { for (int i = 0; i < messages.size(); i++) { if
+ * (messages.get(i).getBody().contentEquals(smsMessage.getBody()) &&
+ * messages.get(i).getSender().contentEquals(smsMessage.getSender())) { return i; } } return -1; }*
+ */
```
Comment (cotom972) [inline projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java]: Not relevant. To be removed after review's done.

File: projects/noloan/app/src/main/java/hasadna/noloan/Messages.java [hunk 4]
```diff
@@ -112,39 +108,46 @@ public void undoSuggestion(SmsMessage smsMessage) {
...
 
-  // Return -1 if no message found
-  public int getIndexById(SmsMessage smsMessage) {
-    for (int i = 0; i < messages.size(); i++) {
-      if (messages.get(i).getId().contentEquals(smsMessage.getId())) {
+  // Search message by sender & body. If no message found, return -1
+  public int searchInboxMessage(SmsMessage smsMessage) {
+    for (int i = 0; i < inboxMessages.size(); i++) {
+      if (inboxMessages.get(i).getBody().contentEquals(smsMessage.getBody())
+          && inboxMessages.get(i).getSender().contentEquals(smsMessage.getSender())) {
```
Comment (cotom972) [inline projects/noloan/app/src/main/java/hasadna/noloan/Messages.java]: -> public int searchMessage(SmsMessage smsMessage, **Messages.getInboxMessages() / Messages.getDbMessage**)

Combine to one function with a list parameter?

Comment (oferb) [reply projects/noloan/app/src/main/java/hasadna/noloan/Messages.java]: Yes, have one private one that gets a list param, and 2 public ones that call it.

File: projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java [hunk 5]
```diff
@@ -45,25 +40,14 @@ public RecyclerViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i
...
   @Override
   public void onBindViewHolder(@NonNull RecyclerViewHolder recyclerViewHolder, int i) {
-    recyclerViewHolder.bind(messages.get(i));
+    recyclerViewHolder.bind(Messages.getInstance().getInboxMessages().get(i));
   }
 
   @Override
   public int getItemCount() {
-    return messages.size();
+    return Messages.getInstance().getInboxMessages().size();
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java]: I think you can shorted `getInstance()` to `get()`.

File: projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java [hunk 6]
```diff
@@ -152,14 +134,23 @@ public void bind(SmsMessage sms) {
       buttonCreateLawsuit.setOnClickListener(
           view -> {
             Intent intentToLawsuitForm = new Intent(view.getContext(), LawsuitActivity.class);
+            // TODO: Pass the whole SmsMessage.proto object to the intent.
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java]: Please describe why this change should be done.

Also below.

File: projects/noloan/app/src/main/java/hasadna/noloan/Messages.java [hunk 7]
```diff
@@ -218,6 +230,26 @@ public void notifyListeners(int index, SmsMessage smsMessage, Type type) {
...
+
+  public List<SmsMessage> getInboxMessages() {
+    return inboxMessages;
+  }
+
+  public void setInboxMessages(List<SmsMessage> inboxMessages) {
+    this.inboxMessages = inboxMessages;
+  }
+
+  public void setContext(Context context) {
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/Messages.java]: Where do you use `setContext()`?

File: projects/noloan/app/src/main/java/hasadna/noloan/lawsuit/LawsuitActivity.java [hunk 8]
```diff
@@ -347,5 +366,39 @@ public void onBackPressed() {
       viewPager.setCurrentItem(viewPager.getCurrentItem() - 1);
     }
   }
+
+  public void updateSharedPreferences() {
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/lawsuit/LawsuitActivity.java]: I think it would be nicer to put this in an `AppSharedPreferences.java` class with static methods.

Comment (cotom972) [reply projects/noloan/app/src/main/java/hasadna/noloan/lawsuit/LawsuitActivity.java]: I thought so too, wasn't sure if it's "enough" to create a dedicated class for it.
Changed it :+1:

File: projects/noloan/app/src/main/java/hasadna/noloan/Messages.java (hunk 7 above)
Comment (cotom972) [reply projects/noloan/app/src/main/java/hasadna/noloan/Messages.java]: I wanted to move `readSmsFromDevice()` from the `MainActivity` to be part of `SmsMessages` functions, it was needed for that,but I ran into a problem. Not sure if that's correct though in terms of structure, I'll raise it up in the channel.
Anyway, removed it for now :+1: .
